pkg/amber/claim.go

```diff
@@ -116,13 +104,21 @@ func validateClaimPredicate(predicate ClaimPredicate) (*ClaimPredicate, error) {
 		}
 	}
 
-	// Verify that ExpiresOn is greater than IssuedOn, if the former is provided.
-	if predicate.Metadata != nil {
-		if predicate.Metadata.ExpiresOn != nil &&
-			predicate.Metadata.ExpiresOn.Before(*predicate.Metadata.IssuedOn) {
-			return nil, fmt.Errorf("expiredOn (%v) is before issuedOn (%v)",
-				*predicate.Metadata.ExpiresOn,
-				*predicate.Metadata.IssuedOn)
+	// Verify that NotBefore is after than IssuedOn (inclusive).
+	if predicate.Validity != nil {
```

Comment:
Can predicate.Validity be nil or would that be an error in itself? I think predicate.Validity is not optional, but maybe I am missing something.

Either way, I think you could check `predicate.Validity != nil` only once.